py4j-python/src/py4j/java_gateway.py

```diff
@@ -1305,7 +1305,8 @@ def stream(self, *args):
             answer, self.gateway_client, self.target_id, self.name)
 
         for temp_arg in temp_args:
-            temp_arg._detach()
+            if hasattr(temp_arg, "_detach"):
```

Comment:
This and the below should be `isinstance(temp_arg, JavaObject)`?

Reply:
Will that `JavaObject` assumption break downstream codes that have customized objects with `_detach` defined?
Seems rare though.

Reply:
I think that's rather a bug if we call `_detach` on arbitrary objects unrelated to `py4j`.

Reply:
Sorry I meant external developers' codes that extend `py4j`, which customized a class with `_detach` method.

Reply:
I think either way is fine. In Py4J both `isinstance` and `hasattr` usages are mixed anyway. Some interfaces like Converter just assumes that it has `convert` method too.